downstairs/src/main.rs

```diff
@@ -387,25 +388,28 @@ async fn do_work(
 /*
  * Look at all the work outstanding for this downstairs and make a list
  * if jobs that are new or are waiting for dependencies.
+ *
  * Once we have that list, walk them and see if any are ready to go.  If
  * so, then do that work.
+ *
+ * Upstairs UUID is passed in so we can skip jobs that aren't for this
```

Comment:
Im trying to decide if this should be a "panic we missed something" kind of issue, or a "yeah, this could happen if things fail in just the wrong way" kind of issue.

If it is the panic kind, then we should panic so we can figure out why and plug the hole.  If it's just a message that is printed, we might miss it.